src/query/graphite/native/builtin_functions.go

```diff
@@ -242,6 +242,40 @@ func timeShift(
 	}, nil
 }
 
+
+// delay shifts all samples later by an integer number of steps. This can be used
+// for custom derivative calculations, among other things. Note: this will pad
+// the early end of the data with NaN for every step shifted. delay complements
+// other time-displacement functions such as timeShift and timeSlice, in that
+// delay is indifferent about the step intervals being shifted.
+func delay(
+	ctx *common.Context,
+	singlePath singlePathSpec,
+	steps int,
+) (ts.SeriesList, error) {
+		input := ts.SeriesList(singlePath)
+		output := make([]*ts.Series, input.Len())
+
+		for i, series := range input.Values {
+			delayedVals := delayValuesHelper(ctx, *series, steps)
+			delayedSeries := ts.NewSeries(ctx, series.Name(), series.StartTime(), delayedVals)
+			renamedSeries := delayedSeries.RenamedTo(fmt.Sprintf("delay(%s,%d)", delayedSeries.Name(), steps))
+			output[i] = renamedSeries
+		}
+		input.Values = output
+		return input, nil
+}
+
+// delayValuesHelper takes a series and returns a copy of the values after
+// delaying the values by `steps` number of steps
+func delayValuesHelper(ctx *common.Context, series ts.Series, steps int ) (ts.Values) {
```

Comment:
nit: Remove the extra space here `steps int )` should be `steps int)`.

also nit: Should be just `ts.Values` not `(ts.Values)` since it's single return type.